haproxy/datadog_checks/haproxy/legacy/haproxy.py

```diff
@@ -192,11 +192,12 @@ def _run_socket_commands(parsed_url, commands):
             output = sock.recv(BUFSIZE)
         sock.close()
 
-        responses = response.split('\n\n')
-        if len(responses) != len(commands) + 1 or responses[len(responses) - 1] != '':
-            raise CheckException("Got a different number of responses than expected")
+        responses = [r.strip() for r in response.split('\n\n') if r.strip()]
```

Comment:
For context, the expected response from the socket is typically this:

```
Name: HAProxy
Version: 1.9.10
Release_date: 2019/08/08
<REDACTED FOR BRIEVETY>
ConnectedPeers: 0
DroppedLogs: 0
BusyPolling: 0

# pxname,svname,qcur,qmax,scur,smax,slim,stot,bin,bout,dreq,dresp,ereq,econ,eresp,wretr,wredis,status,weight,act,bck,chkfail,chkdown,lastchg,downtime,qlimit,pid,iid,sid,throttle,lbtot,tracked,type,rate,rate_lim,rate_max,check_status,check_code,check_duration,hrsp_1xx,hrsp_2xx,hrsp_3xx,hrsp_4xx,hrsp_5xx,hrsp_other,hanafail,req_rate,req_rate_max,req_tot,cli_abrt,srv_abrt,comp_in,comp_out,comp_byp,comp_rsp,lastsess,last_chk,last_agt,qtime,ctime,rtime,ttime,agent_status,agent_code,agent_duration,check_desc,agent_desc,check_rise,check_fall,check_health,agent_rise,agent_fall,agent_health,addr,cookie,mode,algo,conn_rate,conn_rate_max,conn_tot,intercepted,dcon,dses,wrew,connect,reuse,cache_lookups,cache_hits,
public,FRONTEND,,,0,1,2000,71,14443,954415,0,0,0,,,,,OPEN,,,,,,,,,1,2,0,,,,0,2,0,2,,,,0,71,0,0,0,0,,2,2,71,,,0,0,0,0,,,,,,,,,,,,,,,,,,,,,http,,2,2,71,0,0,0,0,,,0,0,
datadog,singleton:8080,0,0,0,0,,0,0,0,,0,,0,0,0,0,no check,1,1,0,,,503,,,1,3,1,,0,,2,0,,0,,,,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,,,,,,,,,127.0.0.1:8080,,http,,,,,,,,0,0,0,,,
datadog,singleton:8081,0,0,0,0,,0,0,0,,0,,0,0,0,0,no check,1,1,0,,,503,,,1,3,2,,0,,2,0,,0,,,,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,,,,,,,,,127.0.0.1:8081,,http,,,,,,,,0,0,0,,,
datadog,otherserver,0,0,0,0,,0,0,0,,0,,0,0,0,0,no check,1,1,0,,,503,,,1,3,3,,0,,2,0,,0,,,,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,,,,,,,,,127.0.0.1:1234,,http,,,,,,,,0,0,0,,,
datadog,BACKEND,0,0,0,1,200,71,14443,954415,0,0,,0,0,0,0,UP,3,3,0,,0,503,0,,1,3,0,,0,,1,2,,2,,,,0,71,0,0,0,0,,,,71,0,0,0,0,0,0,0,,,0,0,0,0,,,,,,,,,,,,,,http,roundrobin,,,,,,,0,0,0,0,0,
anotherbackend,singleton:8080,0,0,0,0,,0,0,0,,0,,0,0,0,0,no check,1,1,0,,,503,,,1,4,1,,0,,2,0,,0,,,,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,,,,,,,,,127.0.0.1:8080,,http,,,,,,,,0,0,0,,,
anotherbackend,singleton:8081,0,0,0,0,,0,0,0,,0,,0,0,0,0,no check,1,1,0,,,503,,,1,4,2,,0,,2,0,,0,,,,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,,,,,,,,,127.0.0.1:8081,,http,,,,,,,,0,0,0,,,
anotherbackend,otherserver,0,0,0,0,,0,0,0,,0,,0,0,0,0,no check,1,1,0,,,503,,,1,4,3,,0,,2,0,,0,,,,0,0,0,0,0,0,,,,,0,0,,,,,-1,,,0,0,0,0,,,,,,,,,,,,127.0.0.1:1234,,http,,,,,,,,0,0,0,,,
```

That is, two groups of lines separated by an empty line — hence splitting on `\n\n`. The problem arises when the second group ends or does not end with a `\n` — so the focus here is on stripping, plus skipping any empty group (which might be the case if an empty line is present in the output).

I'm not sure yet if something changed in HAProxy regarding to this newline formatting, or if this will fix the customer's case at all, but this is the error they're seeing and we checked their data was looking like the above^, so I'm trying this out.